lib/db/mysql/index.js

```diff
@@ -460,6 +460,30 @@ MysqlStore.prototype = {
     });
   },
 
+  purgeExpiredTokens: function purgeExpiredTokens(numberOfTokens, delaySeconds, ignoreClientId){
+    var self = this;
+
+    return self.getClientDevelopers(ignoreClientId)
+      .then(function (ignoreClient) {
+        // This ensures that purgeExpiredTokens can not be called with an invalid ignoreClientId
+      })
+      .then(function () {
+        return self._write('CALL PURGE_EXPIRED_TOKENS(?,?,?,@deleted,@rounds)', [
+          numberOfTokens,
+          delaySeconds,
+          ignoreClientId
+        ]);
+      })
+      .then(function() {
+        logger.debug('purgeExpiredTokens completed');
+      })
+      .catch(function(err){
+        err = new Error('Invalid ignoreClientId, please ensure client exists.');
+        logger.error(err);
+        throw err;
+      });
```

Comment:
nit: in order to void catching error that aren't related to the ignoreClientId, I think this `catch` should be further up in the call chain, e.g.:

```
return getClientDevelopers(ignoreClientId)
  .catch( /*  catch and re-throw errors raised by that specific call */ )
  .then( /* then continue with the other processing stuff */ )
```